tests/Mono.Android-Tests/Xamarin.Android.Net/AndroidMessageHandlerNegotiateAuthenticationTests.cs

```diff
@@ -0,0 +1,133 @@
+using System;
+using System.Net;
+using System.Net.Http;
+using System.Reflection;
+using System.Text;
+using System.Threading.Tasks;
+using Xamarin.Android.Net;
+using NUnit.Framework;
+
+namespace Xamarin.Android.NetTests {
+	[TestFixture]
+	public sealed class AndroidMessageHandlerNegotiateAuthenticationTests
+	{
+		[SetUp]
+		public void SetUp ()
+		{
+			AppContext.SetSwitch ("Xamarin.Android.Net.UseNegotiateAuthentication", true);
```

Comment:
…especially considering that the *docs* state that the way to enable NTLM support is by setting the `$(AndroidUseNegotiateAuthentication)` MSBuild property to True, but our unit tests don't actually use that mechanism.  Instead, the tests use `AppContext.SetSwitch("Xamarin.Android.Net.UseNegotiateAuthentication", true)`.

Offhand:

  1. I'm not entirely certain that things will work *as documented*, by setting `$(AndroidUseNegotiateAuthentication)`=true.
  2. I don't know what happens if two different fragments of code call `AppContext.SetSwitch()` with true *and* false.  Which "wins"?
  3. What happens if `AppContext.SetSwitch()` is used *and* `$(AndroidUseNegotiateAuthentication)` is used?  Which "wins"?

Reply:
I implemented the [feature switch pattern](https://github.com/dotnet/designs/blob/main/accepted/2020/feature-switch.md) we use in .NET. I'll write some docs later today where I'll try to make it clearer.

ad 1) In the Mono.Android-Tests project, the msbuild property didn't have any effect. I'm not sure how exactly the test project build works in Xamarin.Android and how the property is propagated, so I resorted to using the AppContext.SetSwitch at runtime. The property works correctly though in `net7.0-android` projects and the `LinkerTests` test demonstrates it.

ad 2 and 3) The `AppContext.SetSwitch` used at runtime should only work in Debug builds, because in Release builds linker bakes the boolean value in. I'll mention it in the docs as well. The MSBuild property will set the initial value and the `SetSwitch(x)` will override it at runtime, so it will win. The `AppContext` is basically a global thread-safe dictionary instance, so the value will correspond to the last `SetSwitch` call (again, only in Debug builds).

To avoid confusion, I'll mention only the MSBuild property in the docs (in `build-properties.md`) and add a comment to the test to explain why I'm using AppContext.

Reply:
There is a bug - when `$(AndroidUseNegotiateAuthentication)` is set to true, the value is correctly baked in in Release builds but the AppContext switch is not set in Debug builds. That's why the test was failing... I'm working on a fix.